WalletWasabi/Services/UpdateManager.cs

```diff
@@ -0,0 +1,239 @@
+	{
+		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+		{
+			var url = urls.Where(url => url.Contains(".msi")).First();
+			return (url, url.Split("/").Last());
+		}
+		else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+		{
+			var cpu = RuntimeInformation.ProcessArchitecture;
+			if (cpu.ToString() == "Arm64")
+			{
+				var arm64url = urls.Where(url => url.Contains("arm64.dmg")).First();
+				return (arm64url, arm64url.Split("/").Last());
+			}
+			var url = urls.Where(url => url.Contains(".dmg") && !url.Contains("arm64")).First();
+			return (url, url.Split("/").Last());
+		}
+		else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+		{
+			throw new InvalidOperationException("For Linux, get the correct update manually.");
+		}
+		else
+		{
+			throw new InvalidOperationException("OS not recognized, download manually.");
+		}
+	}
+
+	public event EventHandler<UpdateStatus>? UpdateAvailableToGet;
+
+	public string DataDir { get; }
+	public string DownloadsDir { get; }
+	public IHttpClient HttpClient { get; }
+	public bool UpdateOnClose { get; set; }
+
+	public bool InstallNewVersion()
+	{
+		try
+		{
+			var installerPath = Path.Combine(DataDir, "Downloads", InstallerName);
+			ProcessStartInfo startInfo;
+			if (!File.Exists(installerPath))
+			{
+				throw new FileNotFoundException(installerPath);
+			}
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				startInfo = ProcessStartInfoFactory.Make(installerPath, "", true);
+			}
+			else
+			{
+				startInfo = new()
+				{
+					FileName = installerPath,
+					UseShellExecute = true,
+					WindowStyle = ProcessWindowStyle.Normal
+				};
+			}
+
+			using Process? p = Process.Start(startInfo);
+			p!.WaitForExit();
+
+			// Exit code -- Reason
+			//	___________________
+			//	1602	 -- Canceled
+			//	1		 -- Terminated
+			//	0		 -- Finished
+			if (p.ExitCode == 0)
+			{
+				if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+				{
+					// For MacOS, you need to start the process twice, first start => permission denied
+					// TODO: find out why and fix.
+					p.Start();
+					p!.WaitForExit();
+				}
+				else
+				{
+					Logger.LogInfo("Succesfuly installed new version. Deleting installer.");
+					Directory.Delete(DownloadsDir, true);
+					return true;
+				}
+			}
+			else
+			{
+				Logger.LogError("Wasabi installer was terminated.");
+			}
+		}
+		catch (Exception ex)
+		{
+			Logger.LogError("Failed to install latest release. File might be corrupted. Deleting...", ex);
+			Directory.Delete(DownloadsDir, true);
+		}
+		return false;
+	}
+
+	public bool CheckIfInstallerDownloaded()
+	{
+		var folder = new DirectoryInfo(DownloadsDir);
+		if (folder.Exists)
+		{
+			var files = folder.GetFileSystemInfos();
+			if (files.Length == 0)
+			{
+				return false;
+			}
+			return files.Any(file =>
+			{
+				if (file.Name.Contains("Wasabi"))
+				{
+					InstallerName = file.Name;
+					return true;
+				}
+				return false;
+			});
```

Comment:
This smells for me.

```suggestion
			if(files.Any(file => file.Name.Contains("Wasabi"))
			{
			     InstallerName = file.Name;
			     return true;
			}
			return false
```

Reply:
This does not work but did a better update on this.